lib/quicly.c

```diff
@@ -1140,18 +1153,24 @@ static int discard_handshake_context(quicly_conn_t *conn, const quicly_sent_pack
     return 0;
 }
 
-static void apply_peer_transport_params(quicly_conn_t *conn)
+static int apply_peer_transport_params(quicly_conn_t *conn)
 {
+    int ret;
+
     conn->egress.max_data.permitted = conn->super.peer.transport_params.max_data;
-    update_max_streams(&conn->egress.max_streams.uni, conn->super.peer.transport_params.max_streams_uni);
-    update_max_streams(&conn->egress.max_streams.bidi, conn->super.peer.transport_params.max_streams_bidi);
+    if ((ret = update_max_streams(&conn->egress.max_streams.uni, conn->super.peer.transport_params.max_streams_uni)) != 0)
+        return ret;
+    if ((ret = update_max_streams(&conn->egress.max_streams.bidi, conn->super.peer.transport_params.max_streams_bidi)) != 0)
+        return ret;
+
+    return 0;
 }
 
 void quicly_free(quicly_conn_t *conn)
 {
     LOG_CONNECTION_EVENT(conn, QUICLY_EVENT_TYPE_FREE);
 
-    destroy_all_streams(conn, 0);
+    destroy_all_streams(conn, 0, 1);
```

Comment:
```suggestion
    destroy_all_streams(conn, /*err=*/0, /*including_crypto_streams=*/1);
```

Reply:
While I understand the intent, I wonder if it's something we want to maintain or encourage.

I'd argue that it's the type of the information that the compiler knows, and IDEs let's you obtain the information on-the-fly. For example, Xcode shows the list of arguments along with doc-comment, when you press Option+Esc while the mouse cursor is above the name of the function.

Isn't that better? Though admittedly, I do not use Xcode for code reviews (maybe we should).

Reply:
Fair point. I think using Xcode for reviews seems good -- I'm going to try that!